Add procedure, concept, and reference for image mode hosts (#3663)
theforeman/foreman-documentation: guides/common/modules/con_package-mode-and-image-mode-hosts.adoc

> [id="package-mode-and-image-mode-hosts_"]

Reviewer: ```suggestion
[id="package-mode-and-image-mode-hosts"]
```

[id="package-mode-and-image-mode-hosts_"]
= Package mode and image mode hosts

{Project} identifies an image mode host as any system that reports a container image from its bootc status through Subscription Manager. 
It classifies all other hosts as package mode hosts.

Image mode simplifies enterprise operations by enabling development, operations, and solution providers to use the same container-native tools and techniques to manage both applications and the underlying operating system.

In image mode, users can install and manage packages, but with limitations. 
Any packages installed on image mode hosts are temporary and disappear after a reboot. 
Because of this, installing packages is generally discouraged, except for debugging purposes or, in some cases, when applying critical security patches.

> You can display a column in the New UI of the *All Hosts* page that includes a mode indicator. 

Reviewer 2: This is a late code change, but the column is going to be called "Type" instead of having the image mode icon. https://github.com/Katello/katello/pull/11319

Reviewer 2: This is worded without specifically mentioning the name of the column so we could leave it as-is if necessary.

> In the New UI of the *All Hosts* page, you can display the *Type* column, which includes a mode indicator. 

Reviewer 2: Perfect!

> In the New UI of the *All Hosts* page, you can display the *Type* column, which includes a mode indicator. 

Reviewer 3: ```suggestion
On the *All Hosts* page, you can display the *Type* column, which includes a mode indicator. 
```
(The New UI is the default in this version.)

> This column is disabled by default, so enabling it views mode information.

Reviewer 3: ```suggestion
This column is disabled by default.
```
Sufficient

.Image mode hosts
Image mode hosts include additional fields that reflect their boot status.
These fields track the container images used in various scenarios:

* *Booted Image & Digest*: Indicates the container image the host is currently using.
* *Staged Image & Digest*: Indicates the container image the host uses after the next reboot.
* *Available Image & Digest*: Represents the container image cached by running `bootc upgrade --check`.
* *Rollback Image & Digest*: Represents the container image the host reverts to after the next reboot if `Bootc rollback` is applied.

.Additional resources

> For more information on image mode concepts, refer to the https://www.redhat.com/en/blog/image-mode-red-hat-enterprise-linux-quick-start-guide[RHEL Image Mode quick start guide].

Reviewer 3: ```suggestion
* For more information on image mode, see the https://www.redhat.com/en/blog/image-mode-red-hat-enterprise-linux-quick-start-guide[RHEL Image Mode quick start guide].
```
